jsonrpc2/codecs.go

```diff
@@ -26,31 +52,121 @@ var _ Codec = &jsonCodec{}
 // IOCodec returns a Codec that wraps JSON encoding and decoding over IO.
 func IOCodec(rwc io.ReadWriteCloser) *jsonCodec {
```

Comment:
Intentional discrepancy?  Public function is `IOCodec` but it returns a `jsonCodec`?

Reply:
All the *Codec helpers return a Codec interface implementation. In this case, it takes an IO (rwc) and returns a Codec, so it's an IOCodec.

I don't return the Codec interface explicitly to comply with the "accept interfaces, return structs" mantra.

The fact that it's a jsonCodec is because it is _json_ RPC after all. :)

I should probably:
1. Pull all these codecs into a codec sub-package, then could have `jsonrpc2.codec.New(rwc)`
2. Maybe make the jsonCodec struct a public JSONCodec so people can directly provide their own encoders/decoders.

I've been going back and forth on how I wanted the jsonCodec internals to be used, but it might be stable enough now.